code/billy/django_grocery_list/grocery_list/views.py

```diff
@@ -1,6 +1,31 @@
-from django.shortcuts import render
-from django.http import HttpResponse
+from django.shortcuts import render, get_object_or_404
+from django.http import HttpResponseRedirect
+from django.urls import reverse
+from django.utils import timezone
+from .models import GroceryItems
 
 def index(request):
-    return HttpResponse('ok')
+    incomplete_items = GroceryItems.objects.filter(item_completed=False).order_by('created_date')
+    completed_items = GroceryItems.objects.filter(item_completed=True).order_by('-completed_date')
+    context = {
+        'incomplete_items': incomplete_items,
+        'completed_items': completed_items
+    }
+    return render(request, 'grocery_list/index.html', context)
 
+def add(request):
+    item_description = request.POST['item_name']
+    GroceryItems.objects.create(item_name=item_description)
+    return HttpResponseRedirect(reverse('grocery_list:index'))
+
+def complete(request, pk):
```

Comment:
* Need to add functionality to 'incomplete' a grocery item.
* From [Lab 1: Grocery List](https://github.com/PdxCodeGuild/class_swordfish/blob/main/3%20Django/labs/01%20Grocery%20List.md):
```
The user should be able to mark an item complete/incomplete and be able to delete an item.
```